### PR TITLE
JFR CPUPerformance : NetBSD per cpu load

### PR DESCRIPTION
**Pull request by devnexen**
NetBSD subpart. Also build fix while here.

### CODE REVIEW

File: src/hotspot/os/bsd/os_perf_bsd.cpp [hunk 1]
```diff
@@ -132,8 +132,12 @@ size_t length = sizeof(cpu_load_info);
     memcpy(cpu_load_info, &allcpus[which_logical_cpu * CPUSTATES], sizeof(long) * CPUSTATES);
     FREE_C_HEAP_ARRAY(long, allcpus);
 #else
-    /* TODO: NetBSD */
-    return FUNCTIONALITY_NOT_IMPLEMENTED;
+    char mib[18];
+    snprintf(mib, sizeof(mib), "kern.cp_time.%d", which_logical_cpu);
+    mib[sizeof(mib)-1] = 0;
```
Comment (bsdkurt) [inline src/hotspot/os/bsd/os_perf_bsd.cpp]: snprintf guarantees null termination even if buffer is too small for the result so `mib[sizeof(mib)-1] = 0;` is not needed.

While it is extremely unlikely that a system will have more then 9999 logical cpus's, we should probably not impose a limit here. mib array of at least 24 would be better I think (13 chars for 'kern.cp_time.' + 10 chars for max int value + 1 for NULL).

**Review by bsdkurt — CHANGES_REQUESTED** [review]
Thanks again for this. Minor nits below. Also could you just include the os_perf_bsd.cpp change in this pull request and send the os_posix.cpp one directly to `battleblow:bsd-port` branch? It looks like that's a netbsd build fix unrelated to CPUPerformance.

File: src/hotspot/os/bsd/os_perf_bsd.cpp (hunk 1 above)
Comment (devnexen) [reply src/hotspot/os/bsd/os_perf_bsd.cpp]: Yes I wanted to sound "reasonable" in term of possible cpu support on NetBSD :-) but I can change no worries.

**Comment by bsdkurt** [comment]
Hi David,

os_perf_bsd.cpp looks good to me now. Did you see my request to move the os_posix.cpp change of this pull request and to make a pull request on the battleblow:bsd-port branch for it?

Thanks,
-Kurt

**Comment by devnexen** [comment]
Ah no ... doing it.

**Pull request merged**